detect/detect.go

```diff
@@ -252,6 +252,13 @@ func (d *Detector) Detect(fragment Fragment) []report.Finding {
 	decoder := NewDecoder()
 
 	for {
+		select {
+		case <-ctx.Done():
```

Comment:
Is using the `chan` better than checking `ctx.Err() != nil`? Switch with an empty default feels weird.
https://www.reddit.com/r/golang/comments/1e1wdvy/comment/lcxfz14/